### PR TITLE
[viki] Add support for shows

### PR DESCRIPTION
**Pull request by ping**
Add support for shows, example: http://www.viki.com/tv/50c-boys-over-flowers

### CODE REVIEW

File: youtube_dl/extractor/viki.py [hunk 1]
```diff
@@ -145,3 +145,35 @@ def _get_subtitles(self, video_id, info_webpage):
...
+
+    def _real_extract(self, url):
+        show_id = self._match_id(url)
+        show_page = self._download_webpage(url, show_id, 'Download show page')
+
+        title = self._og_search_title(show_page)
+        description = self._og_search_description(show_page)
+
+        show_json = self._download_json(
+            'http://api.viki.io/v4/containers/%s/episodes.json?app=100000a&per_page=999&sort=number&direction=asc' % show_id,
```
Comment (dstftw) [inline youtube_dl/extractor/viki.py]: Maximum `per_page` is 25 according [documentation](http://dev.viki.com/v4/overview/#paging). However, I've managed to fetch 50 (for test http://www.viki.com/tv/1354c-poor-nastya-complete). You'll need to implement proper paging approach.

**Comment by dstftw** [comment]
Sometimes there are no episodes, but clips (http://www.viki.com/tv/2500c). May probably reasonable to download them in this case. Or even always download both if present.

**Comment by ping** [comment]
Thanks for the review. I've gone with using 25 per_page, since it's the official number.

**Comment by dstftw** [comment]
Can you make it as variable, e.g. `_PER_PAGE`?

File: youtube_dl/extractor/viki.py [hunk 2]
```diff
@@ -145,3 +145,46 @@ def _get_subtitles(self, video_id, info_webpage):
                 'ext': 'vtt',
             }]
         return res
+
+
+class VikiShowIE(InfoExtractor):
+    IE_NAME = 'viki:show'
```
Comment (dstftw) [inline youtube_dl/extractor/viki.py]: I think this should be renamed to channel or container (for what `c` stands for).

**Comment by ping** [comment]
By _PER_PAGE as a variable, do you mean just as a regular class variable? Or is there some other mechanism that I can hook into?

```
class VikiShowIE(InfoExtractor):
    _PER_PAGE = 25
    # -- snip --

    def _real_extract(self, url):
        # -- snip --
        for video_type in ['episodes', 'clips']:
            json_url = 'http://api.viki.io/v4/containers/%s/%s.json?app=100000a&per_page=%d&sort=number&direction=asc&with_paging=true&page=1' % (show_id, video_type, self._PER_PAGE)
```


 ``show_json['pagination']['next'] `` is a fully formed url, so there's no need to build the url with an incremental ``page`` number

File: youtube_dl/extractor/viki.py (hunk 2 above)
Comment (ping) [inline youtube_dl/extractor/viki.py]: Will change this to channel

**Comment by dstftw** [comment]
Just a variable would be ok. There are some mechanisms if you still want:
https://github.com/rg3/youtube-dl/blob/master/youtube_dl/utils.py#L1402-L1471
https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/vimeo.py#L648-L665
https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/zdf.py#L124-L153

> ``show_json['pagination']['next'] `` is a fully formed url, so there's no need to build the url with an incremental ``page`` number

Sorry, missed that.

**Comment by ping** [comment]
Changed.

**Comment by dstftw** [comment]
Thanks. Do you want to be credited as one of the authors of youtube-dl? If so, under what name?

**Comment by ping** [comment]
That would be nice. Under "Ping O."

**Pull request merged**